README.md

```diff
@@ -0,0 +1,104 @@
+- [Vitess](https://vitess.io/docs/14.0/get-started/) v14.0
+- [Docker](https://www.docker.com/) and [docker-compose](https://docs.docker.com/compose/)
+
+### How to build it
+
+Run `make build`.
+
+### Testing
+
+Run `make test` to run all the unit and integration tests, which require Docker and Compose V2 to be installed and running. The command will handle starting and stopping docker containers for you.
+
+## Source
+
+The Vitess Source Connector connects to a VTGate with the provided `address`, `keyspace` and `tabletType` and starts creating records for each change detected in a table.
+
+Upon starting, the source takes a snapshot of a given table in the database, then switches into CDC mode. In CDC mode, the plugin reads events from a VStream. In order for this to work correctly, binlog must be enabled on MySQL instances you use with Vitess.
+
+### Snapshot Capture
+
+When the connector first starts, snapshot mode is enabled. The connector reads all rows of a table in batches via SELECT with [keyset pagination](https://use-the-index-luke.com/no-offset) by `orderingColumn`, limiting the rows by `batchSize` and ordering them by the `orderingColumn`. Once all rows in that initial snapshot are read the connector switches into CDC mode.
+
+### Change Data Capture
+
+This connector implements CDC features for Vitess by connecting to a VStream that listens to changes in the configured table. Every detected change is converted into a record and returned in the call to `Read`. If there is no available record when `Read` is called, it blocks until a record becomes available or the connector receives a stop signal.
+
+The connector in the CDC mode retrieves all the available shards from Vitess and tracks changes from all of them. If a reshard occurs, the connector will see the change and will listen for events from the new shards.
+
+### Position handling
+
+The connector goes through two modes.
+
+**Snapshot** mode. The position contains `keyspace` and a value of the last processed element of an ordering column you chose. This means that the ordering column must contain unique values.
+
+Example of the Snapshot position:
+
+```json
+{
+  "mode": "snapshot",
+  "keyspace": "test",
+  "last_processed_element_value": 1
+}
+```
+
+**CDC** mode. The position in this mode contains the same fields as in the Snapshot mode plus a list of shards (and their unique shard transaction identifiers, gtid) of a `keyspace` you chose. The connector retrieves all the available shards at startup and watches them for changes.
+
+Example of the CDC position:
+
+```json
+{
+  "mode": "cdc",
+  "keyspace": "test",
+  "last_processed_element_value": 1,
+  "shard_gtids": [
+    {
+      "keyspace": "test",
+      "shard": "-80",
+      "gtid": "MySQL56/36a89abd-978f-11eb-b312-04ed332e05c2:1-265"
+    },
+    {
+      "keyspace": "test",
+      "shard": "80-",
+      "gtid": "MySQL56/36a89abd-978f-11eb-b312-04ed332e05c2:1-265"
+    }
+  ]
+}
+```
+
+### Configuration Options
```

Comment:
Does SDK provide the default options for Source?
I see a TODO comment for the `DefaultSourceMiddleware`: https://github.com/ConduitIO/conduit-connector-sdk/blob/main/source_middleware.go#L26.
Or perhaps this is done in some other branch/version of SDK?

Reply:
That's a good call out, it's not implemented yet. I was thinking that we should add it because we use it, but since it's not implemented it doesn't make sense.